sklearn/impute/_iterative.py

```diff
@@ -565,8 +565,8 @@ def fit_transform(self, X, y=None):
 
         self.imputation_sequence_ = []
 
-        self._min_value = -np.inf if self.min_value is None else self.min_value
-        self._max_value = np.inf if self.max_value is None else self.max_value
+        self._min_value = self.min_value
```

Comment:
You should not do that. In a couple of lines below, you need to change the first occurrence of `self._min_value` to `self.min_value`.